saleor/graphql/channel/types.py

```diff
@@ -1,7 +1,49 @@
 from graphene import relay
+from graphene.types.resolver import get_default_resolver
+from graphene_django import DjangoObjectType
 
 from ...channel import models
 from ..core.connection import CountableDjangoObjectType
+from ..meta.types import ObjectWithMetadata
+from ..translations.resolvers import resolve_translation
+from . import ChannelContext
+
+
+class ChannelContextType(DjangoObjectType):
```

Comment:
This could also be a generic:

```python
N = TypeVar("N")

class ChannelContextType(DjangoObjectType, Generic[N]):
    ...

    @staticmethod
    def resolver_with_context(
        attname, default_value, root: ChannelContext[N], info, **args
    ):
        ...

    # etc.
```